### PR TITLE
feat(no-ref): redesign

### PR DESCRIPTION
**Pull request by andriikamaldinov1**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/JsDaddy/ngx-copypaste/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/app/cards/cards.component.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<section class="content column bg-full-white">
+    <div
+        class="card"
+        *ngFor="let card of cardDocs; let i = index; trackBy: trackByPath"
+        [ngClass]="activeCardId === card.id | color"
```
Comment (GlebChiz) [inline src/app/cards/cards.component.html:5]: Add this logic to pipe color
Should be something like that activeCardId | color : card.id

File: src/app/cards/cards.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public activeCardId = 1;
+
+    public readonly trackByPath = inject(TrackByService).trackBy('id');
+
+    private readonly scrollService = inject(ScrollService);
+    private readonly router = inject(Router);
+
+    public ngAfterViewInit(): void {
+        fromEvent(document, 'scroll')
+            .pipe(takeUntil(this.unsubscribe$$), debounceTime(100))
```
Comment (GlebChiz) [inline src/app/cards/cards.component.ts:54]: Always use the takeUntil operator last
debounceTime(100), ....., takeUntil(this.unsubscribe$$)

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,49 +1,38 @@
...
-  <div>
-    <h2>Lorem ipsum</h2>
-  </div>
-</div>
-<button  (click)="cp.copy()" >Copy</button>`;
+    public card: {
+        docs: ICard[];
+    } = {
+        docs: ComDocs,
+    };
```
Comment (lachesse) [inline src/app/app.component.ts:32]: what is the object for? why not just docs variable?

File: src/app/cards/cards.component.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<section class="content column bg-full-white">
+    <div
+        class="card"
+        *ngFor="let card of cardDocs; let i = index; trackBy: trackByPath"
```
Comment (lachesse) [inline src/app/cards/cards.component.html:4]: is an index needed here?

File: src/app/cards/cards.interface.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export interface ICard {
+    header: string;
+    text: string;
+    code: string;
+    id: number;
+    anchor: string;
+    input?: boolean;
+    textarea?: boolean;
```
Comment (lachesse) [inline src/app/cards/cards.interface.ts:8]: Do you really need two fields, input and textarea? I think one is enough, isTextarea or isInput

Also, you have an empty text field everywhere, do you need it?

Comment (andriikamaldinov1) [reply src/app/cards/cards.interface.ts]: With input you are right, i needed another text font size

Comment (lachesse) [reply src/app/cards/cards.interface.ts]: I meant the "text" field in the interface (text: string), as I understand it, in all cases you have it empty. Do you need it?

File: src/app/app.component.scss [hunk 6]
```diff
@@ -1,34 +1,26 @@
+@import 'src/styles';
+:host {
+    display: block;
+    height: 100%;
+}
 .content-wrapper {
```
Comment (xakeppok) [inline src/app/app.component.scss:6]: use it in correct structure in :host

File: src/app/app.component.scss [hunk 7]
```diff
@@ -1,34 +1,26 @@
...
         }
     }
 }
 
-@media only screen and (max-width: 980px) {
-    .main-container {
-        max-width: 100%;
-        min-width: 100%;
+@media screen and (max-width: $desktopWidth) {
+    .content-wrapper {
```
Comment (xakeppok) [inline src/app/app.component.scss:18]: use same structure as you using in not media query

File: src/app/cards/cards.component.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    @ViewChildren('cards') public cards!: QueryList<ElementRef>;
+
+    public activeCardId = 1;
+
+    public readonly trackByPath = inject(TrackByService).trackBy('id');
+
+    private readonly scrollService = inject(ScrollService);
+    private readonly router = inject(Router);
+
+    public ngAfterViewInit(): void {
```
Comment (xakeppok) [inline src/app/cards/cards.component.ts:52]: i see it in the ngx mask. please create reusable service for this

File: src/app/cards/cards.interface.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+export interface ICard {
+    header: string;
+    code: string;
+    id: number;
+    anchor: string;
+    isInput?: boolean;
+    text12?: string;
```
Comment (xakeppok) [inline src/app/cards/cards.interface.ts:7]: what doe '12','14', etc means?

File: src/app/shared/copied/copied.component.html [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<button class="row bg-full-white black ai-start span copy-btn">
+    <span>Copy</span>
+    <img class="copy ais-c" [ngSrc]="'copied' | asset" width="20" height="10" alt="copied" />
```
Comment (xakeppok) [inline src/app/shared/copied/copied.component.html:3]: please use correct paths for assets. look how it resolved in company site

**Pull request merged**